### PR TITLE
[LoRA] relax lora loading logic

### PR DESCRIPTION
**Pull request by sayakpaul**
Internal discussion thread: https://huggingface.slack.com/archives/C03UQJENJTV/p1692016705461479.

Tested with the following:

```python
import torch
from diffusers import StableDiffusionPipeline

pipeline = StableDiffusionPipeline.from_pretrained(
    "gsdf/Counterfeit-V2.5",
    safety_checker=None,
    use_safetensors=True,
)
pipeline.load_lora_weights("sayakpaul/civitai-light-shadow-lora")
```

## Todo

- [ ] Add an integration test

### CODE REVIEW

File: src/diffusers/loaders.py [hunk 1]
```diff
@@ -1074,6 +1086,28 @@ def lora_state_dict(
...
+        else:
+            files_in_repo = model_info(pretrained_model_name_or_path_or_dict).siblings
+            targeted_files = [f.rfilename for f in files_in_repo if f.rfilename.endswith(file_extension)]
+
+        if len(targeted_files) == 0:
+            return
+
+        targeted_files = list(filter(lambda x: "scheduler" not in x and "optimizer" not in x, targeted_files))
+        if len(targeted_files) > 1:
+            raise ValueError(
```
Comment (apolinario) [inline src/diffusers/loaders.py:1105]: WDYT if this is a warning not a `raise`, and we pick the first one? Not a strong opinion tho.

Comment (sayakpaul) [reply src/diffusers/loaders.py:1105]: Better to raise an error than leaving it silent IMO. This promotes a sense to structuring the Hub repos too IMO.

Comment (apolinario) [reply src/diffusers/loaders.py:1105]: Fair. I thought about doing just a warning and still trying the best guess, like "we picked this, if you meant another one, improve your repo"

**Review by apolinario — APPROVED** [review]

File: src/diffusers/loaders.py (hunk 1 above)
Comment (apolinario) [reply src/diffusers/loaders.py:1105]: Example use-case where this is useful: https://huggingface.co/Linaqruf/pastel-anime-xl-lora/tree/main

Comment (sayakpaul) [reply src/diffusers/loaders.py:1105]: Sure. But I think we need to meet at a common point and I would prefer the current design to promote what I said in the earlier comment.

**Comment by sayakpaul** [comment]
@williamberman could you give this a look?

**Review by patrickvonplaten — APPROVED** [review]
Nice! Looks good to me

**Comment by sayakpaul** [comment]
@patrickvonplaten possible to give another thorough look. Had to change a few minor things to make everything work.

**Review by williamberman — APPROVED** [review]
Nice!

**Pull request merged**